### PR TITLE
gh-125470: Fix warning in `Python/generated_cases.c.h`

### PR DESCRIPTION
**Pull request by sobolevn**
Refs https://github.com/python/cpython/pull/125324

* Issue: gh-125470

### CODE REVIEW

File: Python/bytecodes.c [hunk 1]
```diff
@@ -611,7 +611,9 @@ dummy_func(
         // specializations, but there is no output.
         // At the end we just skip over the STORE_FAST.
         op(_BINARY_OP_INPLACE_ADD_UNICODE, (left, right --)) {
+        #ifndef NDEBUG
```
Comment (Fidget-Spinner) [inline Python/bytecodes.c:614]: Out of curiosity, what's the difference between this an `Py_DEBUG` ?

Comment (sobolevn) [reply Python/bytecodes.c:614]: See https://github.com/python/cpython/blob/5dac0dceda9097d46a0b5a6ad7c927e002c6c7a5/Include/cpython/object.h#L372-L385 

And https://github.com/python/cpython/blob/5dac0dceda9097d46a0b5a6ad7c927e002c6c7a5/Doc/using/configure.rst#L666-L667

:)

Comment (Fidget-Spinner) [reply Python/bytecodes.c:614]: Ok I trust you on this that it's the right choice, but I still don't quite understand why normal `Py_DEBUG` doesn't cut it. The assert below is a normal assert, not the `_PyObject_ASSERT`?

Comment (sobolevn) [reply Python/bytecodes.c:614]: Hm, now I start to question myself with this one :)
I will double check which one is better.

Comment (sobolevn) [reply Python/bytecodes.c:614]: I double checked: `Py_DEBUG` is used for code that should only be present when `--with-pydebug` is set. But `NDEBUG` is used when you want assertions to work. So, this is correct :)

**Review by Fidget-Spinner — APPROVED** [review]

File: Python/bytecodes.c (hunk 1 above)
Comment (Fidget-Spinner) [reply Python/bytecodes.c:614]: Thanks for checking!

**Comment by sobolevn** [comment]
Failures do not look related:

```
======================================================================
FAIL: test_timerfd_poll (test.test_os.TimerfdTests.test_timerfd_poll)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/cpython/cpython/Lib/test/test_os.py", line 4391, in test_timerfd_poll
    self.check_timerfd_poll(False)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^
  File "/home/runner/work/cpython/cpython/Lib/test/test_os.py", line 4387, in check_timerfd_poll
    self.assertGreater(dt, total_time / sec_to_nsec - self.CLOCK_RES)
    ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 0.49887036200016155 not greater than 0.499

----------------------------------------------------------------------
```

I will merge this, thanks everyone!

**Pull request merged**